src/module/canvas/token/object.ts

```diff
@@ -297,57 +297,25 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
         }
 
-        const distance = {
-            xy: measureDistanceRect(this.bounds, target.bounds, { reach }),
-            xz: 0,
-            yz: 0,
-        };
-
         const selfElevation = this.document.elevation;
         const targetElevation = target.document.elevation;
-        if (selfElevation === targetElevation || !this.actor || !target.actor) return distance.xy;
+        if (selfElevation === targetElevation || !this.actor || !target.actor) {
+            return measureDistanceRect(this.bounds, target.bounds, { reach });
+        }
 
         const [selfDimensions, targetDimensions] = [this.actor.dimensions, target.actor.dimensions];
-        if (!(selfDimensions && targetDimensions)) return distance.xy;
+        if (!(selfDimensions && targetDimensions)) return measureDistanceRect(this.bounds, target.bounds, { reach });
 
         const gridSize = canvas.dimensions.size;
         const gridDistance = canvas.dimensions.distance;
 
-        const xzPlane = {
-            self: new PIXI.Rectangle(
-                this.bounds.x,
-                Math.floor((selfElevation / gridDistance) * gridSize),
-                this.bounds.width,
-                Math.floor((selfDimensions.height / gridDistance) * gridSize)
-            ),
-            target: new PIXI.Rectangle(
-                target.bounds.x,
-                Math.floor((targetElevation / gridDistance) * gridSize),
-                target.bounds.width,
-                Math.floor((targetDimensions.height / gridDistance) * gridSize)
-            ),
-        };
-        distance.xz = measureDistanceRect(xzPlane.self, xzPlane.target, { reach });
-
-        const yzPlane = {
-            self: new PIXI.Rectangle(
-                this.bounds.y,
-                Math.floor((selfElevation / gridDistance) * gridSize),
-                this.bounds.height,
-                Math.floor((selfDimensions.height / gridDistance) * gridSize)
-            ),
-            target: new PIXI.Rectangle(
-                target.bounds.y,
-                Math.floor((targetElevation / gridDistance) * gridSize),
-                target.bounds.height,
-                Math.floor((targetDimensions.height / gridDistance) * gridSize)
-            ),
-        };
-        distance.yz = measureDistanceRect(yzPlane.self, yzPlane.target, { reach });
-
-        const hypotenuse = Math.sqrt(Math.pow(distance.xy, 2) + Math.pow(Math.max(distance.xz, distance.yz), 2));
-
-        return Math.floor(hypotenuse / gridDistance) * gridDistance;
+        return measureDistanceRect(this.bounds, target.bounds, {
+            reach,
+            elevation0: Math.floor((selfElevation / gridDistance) * gridSize),
+            height0: Math.floor((selfDimensions.height / gridDistance) * gridSize),
+            elevation1: Math.floor((targetElevation / gridDistance) * gridSize),
+            height1: Math.floor((targetDimensions.height / gridDistance) * gridSize),
+        });
```

Comment:
This signature is kind of silly: 0 and 1 correspond with `this` and `target`

Reply:
should I rename the signature or pass this and target instead?

Reply:
Renaming 0 and 1 to self and target doesn't make it less silly :stuck_out_tongue:

Reply:
Pass the pairs of tokens instead

Reply:
got it. moved :)